e2e/test-runner/runner_test.go

```diff
@@ -16,20 +29,134 @@ func TestBasicSetup(t *testing.T) {
+	{
+		outputStr := tester.AssertWslCommand("whoami")
+		notRoot := strings.Fields(outputStr)[0]
+		if notRoot == "root" {
+			tester.Logf("%s", outputStr)
+			tester.Fatal("Default user should not be root.")
+		}
 	}
 
 	// Subiquity either installs or marks for installation the relevant language packs for the chosen language.
-	outputStr = tester.AssertWslCommand("apt-mark", "showinstall", "language-pack\\*")
-	packs := strings.Fields(outputStr)
-	tester.Logf("%s", outputStr) // I'd like to see what packs were installed in the end, just in case.
-	if len(packs) == 0 {
-		tester.Fatal("At least one language pack should have been installed or marked for installation, but apt-mark command output is empty.")
+	{
+		outputStr := tester.AssertWslCommand("apt-mark", "showinstall", "language-pack\\*")
+		packs := strings.Fields(outputStr)
+		tester.Logf("%s", outputStr) // I'd like to see what packs were installed in the end, just in case.
+		if len(packs) == 0 {
+			tester.Fatal("At least one language pack should have been installed or marked for installation, but apt-mark command output is empty.")
+		}
+	}
+
+	// Proper release. Ensures the right tarball was used as rootfs
+	{
+		expectedRelease := func() string {
+			if *distroName == "Ubuntu-Preview" || *distroName == "UbuntuDev.WslID.Dev" {
+				return "Ubuntu Kinetic Kudu (development branch)"
+			}
+			if *distroName == "Ubuntu" {
+				return "Ubuntu 22.04.1 LTS"
+			}
+			if *distroName == "Ubuntu22.04LTS" {
+				return "Ubuntu 22.04.1 LTS"
+			}
+			if *distroName == "Ubuntu20.04LTS" {
+				return "Ubuntu 20.04.5 LTS"
+			}
+			if *distroName == "Ubuntu18.04LTS" {
+				return "Ubuntu 18.04.6 LTS"
+			}
+			return ""
+		}()
+		if len(expectedRelease) == 0 {
+			tester.Logf("Unknown Ubuntu release corresponding to distro name '%s'", *distroName)
+			tester.Fatal("Unexpected value provided via --distro-name")
+		}
+		outputStr := tester.AssertWslCommand("cat", "/etc/os-release")
+		cfg, err := ini.Load([]byte(outputStr))
+		if err != nil {
+			tester.Logf("Contents of /etc/os-release:\n%s", outputStr)
+			tester.Fatal("Failed to parse ini file")
+		}
+
+		release, err := cfg.Section(ini.DefaultSection).GetKey("PRETTY_NAME")
+		if err != nil {
+			tester.Logf("Contents of /etc/os-release:\n%s", outputStr)
+			tester.Fatal("Failed to find PRETTY_NAME")
+		}
+
+		if release.String() != expectedRelease {
+			tester.Logf("Contents of /etc/os-release:\n%s", outputStr)
+			tester.Logf("Parsed release:   %s", release.String())
+			tester.Logf("Expected release: %s", expectedRelease)
+			tester.Fatal("Unexpected release string")
+		}
+	}
+
+	// Systemd enabled.
+	{
+		getSystemdStatus := func() string {
+			outputStr := tester.AssertWslCommand("bash", "-ec", "systemctl is-system-running || exit 0")
+			return strings.TrimSpace(outputStr)
+		}
+
+		status := getSystemdStatus()
+		for status == "starting" {
+			time.Sleep(1 * time.Second)
+			status = getSystemdStatus()
+		}

```

Comment:
If we only want to check whether systemd was enabled or not we could accept `starting` as a valid answer and move on.

If, instead, we want to make sure systemd is really running, than that's fine. An alternative would be running once the command `systemctl is-system-running --wait` (since v240). No (raw) loops. 😅 . Both approaches are susceptible to failure if D-Bus daemon itself has not yet started, because the systemctl command would fail to connect to D-Bus. I don't know how feasible that is to happen on CI, but we should consider that path otherwise we may turn into brittle tests for unknown reasons. We can run `busctl list --no-pager` first to ensure D-Bus is running before attempting to block on systemd just in case. That command will fail fast if D-Bus is not running.

All tests I did with WSL, though, seem to reveal that WSL waits on systemd before running the systemctl command. I don't think we should rely heavily on that, though.

Reply:
Better safe than sorry. I added both suggestions.

By using `systemctl is-system-running --wait` instead of directly accepting `starting` we also have the implicit check that it does not fail during startup.